packages/services/storage/src/index.ts

```diff
@@ -3975,6 +3975,58 @@ export async function createStorage(
 
       return DocumentCollectionDocumentModel.parse(result);
     },
+    async getPreflightScript(args) {
+      const result = await pool.maybeOne(sql`/* getPreflightScript */
+      SELECT "id"
+           , "source_code"         as "sourceCode"
+           , "target_id"           as "targetId"
+           , "created_by_user_id"  as "createdByUserId"
+           , to_json("created_at") as "createdAt"
+           , to_json("updated_at") as "updatedAt"
+      FROM "document_preflight_scripts"
+      WHERE "target_id" = ${args.targetSlug}
+      `);
+
+      return result && PreflightScriptModel.parse(result);
+    },
+
+    async createPreflightScript(args) {
+      const result = await pool.maybeOne(sql`/* createPreflightScript */
+      INSERT INTO "document_preflight_scripts" ( "source_code"
+                                               , "target_id"
+                                               , "created_by_user_id")
+      VALUES (${args.sourceCode},
+              ${args.targetSlug},
+              ${args.createdByUserId})
+      RETURNING
+          "id"
+          , "source_code" as "sourceCode"
+          , "target_id" as "targetId"
+          , "created_by_user_id" as "createdByUserId"
+          , to_json("created_at") as "createdAt"
+          , to_json("updated_at") as "updatedAt"
+      `);
+
+      return PreflightScriptModel.parse(result);
+    },
+    async updatePreflightScript(args) {
+      const result = await pool.maybeOne(sql`/* updatePreflightScript */
+      UPDATE
+          "document_preflight_scripts"
+      SET "source_code" = ${args.sourceCode}
+        , "updated_at"  = NOW()
+      WHERE "id" = ${args.id}
+      RETURNING
+          "id"
+          , "source_code" as "sourceCode"
+          , "target_id" as "targetId"
+          , "created_by_user_id" as "createdByUserId"
+          , to_json("created_at") as "createdAt"
+          , to_json("updated_at") as "updatedAt"
+      `);
+
+      return result && PreflightScriptModel.parse(result);
+    },
```

Comment:
We want to avoid having this file grow infinitely, so instead you can add these methods and SQL queries within the preflight script provider file.

Reply:
moved